basket/news/backends/ctms.py

```diff
@@ -275,13 +275,15 @@ def to_vendor(data, existing_data=None):
         if isinstance(newsletters, dict):
             # Detect unsubscribe all
             optout = data.get("optout", False) or False
-            if (
-                optout
-                and (not any(newsletters.values()))
-                and (set(valid_slugs) == set(newsletters.keys()))
-            ):
+            if optout:
                 # When unsubscribe all is requested, let CTMS unsubscribe from all
                 output = "UNSUBSCRIBE"
+                # Set has_opted_out_of_email for email
+                if "email" not in ctms_data.keys():
+                    ctms_data["email"] = {}
+                elif ctms_data["email"] is None:
+                    ctms_data["email"] = {}
+                ctms_data["email"]["has_opted_out_of_email"] = True
```

Comment:
I'm surprised this is needed, it should have been set up at line 255. But, since there are no changes to tests, Is suspect it does nothing.

Reply:
I believe you are correct. After this PR, I went back to re-read some of the code and tests. I quickly drafted a better PR I believe here///: https://github.com/mozmeao/basket/pull/709

Reply:
Removed this logic in the new pr